fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java

```diff
@@ -597,6 +599,16 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     @VarAttr(name = CBO_PRUNE_SHUFFLE_COLUMN_RATE, flag = VariableMgr.INVISIBLE)
     private double cboPruneShuffleColumnRate = 0.1;
 
+    /**
+     * If set to true, FE will choose backends on the same host which can be used for better dev because
+     * some cases can only be reproduced on the distributed env.
+     * <p>
+     * NOTE: This config should not be used on the production because data's replicas on the
+     * same host may cause data lost when the host is broken down.
+     */
+    @VarAttr(name = ENABLE_REPLICAS_ON_THE_SAME_HOST, flag = VariableMgr.INVISIBLE)
+    private boolean enableReplicasOnTheSameHost = false;
```

Comment:
Because this will be used by many daemon thread, which don't have connectcontext and seesion variables. So changing this variable can't affect. 
For example when scheduler do balance, it runs in a daemon thread without session variable.
So I think it should be a configure which can be seen by every thread.

Reply:
Thanks for your replies. Let me think it.